daemon/api_systems.go

```diff
@@ -104,19 +105,55 @@ func getAllSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 // wrapped for unit tests
-var deviceManagerSystemAndGadgetInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, error) {
-	return dm.SystemAndGadgetInfo(systemLabel)
+var deviceManagerSystemAndGadgetAndKernelInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, *snap.Info, error) {
+	return dm.SystemAndGadgetAndKernelInfo(systemLabel)
+}
+
+// wrapped for unit tests
+var deviceManagerEncryptionSupportInfo = func(dm *devicestate.DeviceManager, model *asserts.Model, kernelInfo *snap.Info, gadgetInfo *gadget.Info) (devicestate.EncryptionSupportInfo, error) {
+	return dm.EncryptionSupportInfo(model, kernelInfo, gadgetInfo)
+}
+
+func storageEncryption(encInfo *devicestate.EncryptionSupportInfo) *client.StorageEncryption {
+	storageEnc := &client.StorageEncryption{
+		StorageSafety: string(encInfo.StorageSafety),
+		Type:          string(encInfo.Type),
+	}
+	if encInfo.Disabled {
+		storageEnc.StorageSafety = "disabled"
```

Comment:
The string values, should they maybe be const values?

Reply:
Thanks, I added a StorageEncryptionSupport now.